### PR TITLE
sigmac to STIX enhancements

### PR DESCRIPTION
**Pull request by dennispo**
Few enhancements to STIX compiler:
1. Refactored config files:
   - `stix2.0` is a set of official mappings. Later, `stix2.1` can be added.
   - `stix-shifter` is a set of custom mappings supported in stix-shifter project: https://github.com/opencybersecurityalliance/stix-shifter
   - `stix-custom` us a set of mappings that are not mapped in any system currently.
2. Aligned `stix-shifter` mappings to a new naming convention in stix-shifter project.
3. Removed forced mapping to `x-sigma` field, in case where no mapping was found - it is not supported by any system
4. Added fields sorting for more consistent rules output

### CODE REVIEW

**Comment by dennispo** [comment]
The PR is marked as Draft while there is an ongoing conversation in `STIX-shifter` channel about the proper naming.

**Comment by dennispo** [comment]
Meanwhile, I have few ideas for improving sigma2stix compiler and its configuration. @Neo23x0 should it be discussed in issues?

**Comment by dennispo** [comment]
Regarding the naming issue:
All custom stix-shifter fields moved to stix-shifter.yml configuration and updated with the newest stix-shifter naming convention:
- `x-ibm-host` renamed to `x-oca-asset`
- `x-ibm-event` renamed to `x-oca-event`
- `x-ibm-finding` remained unchanged

File: tools/config/stix2.0.yml [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  HomeDirectory:
+    - directory:path
+  Image:
+    - process:binary_ref.name
+  ImageLoadedTempPath:
+    - process:extensions.'windows-service-ext'.service_dll_refs[*].x_temp_path
+  ImageName:
+    - process:binary_ref.name
+  ImagePath:
+    - binary_ref.parent_directory_ref.pat.name
```
Comment (barvhaim) [inline tools/config/stix2.0.yml:184]: `.path.` instead of `.pat`

File: tools/config/stix-custom.yml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    - x-windows:sharename
+  SharePath:
+    - x-windows:sharepath
+  Signature:
+    - x-windows:signature
+  SignatureStatus:
+    - x-windows:signaturestatus
+  Signed:
+    - x-windows:signed
+  SourceImage:
```
Comment (barvhaim) [inline tools/config/stix-custom.yml:79]: possible use of `process:binary_ref.name`

File: tools/config/stix-custom.yml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    - x-windows:startmodule
+  TargetAccountSecurityID:
+    - x-windows:targetaccountsecurityid
+  TargetComputerDomain:
+    - x-windows:targetcomputerdomain
+  TargetComputerName:
+    - x-windows:targetcomputername
+  TargetDetails:
+    - x-windows:targetdetails
+  TargetImage:
```
Comment (barvhaim) [inline tools/config/stix-custom.yml:99]: possible use of `process:binary_ref.name`

**Comment by barvhaim** [comment]
@dennispo looks good, I've added a few comments in code, 
@Neo23x0 do you have any comments for the changes? I see it open for long time

File: tools/config/stix2.0.yml (hunk 1 above)
Comment (dennispo) [reply tools/config/stix2.0.yml]: Good catch. Fixed.

File: tools/config/stix-custom.yml (hunk 2 above)
Comment (dennispo) [reply tools/config/stix-custom.yml]: Done

File: tools/config/stix-custom.yml (hunk 3 above)
Comment (dennispo) [reply tools/config/stix-custom.yml]: Done

**Pull request merged**